cardano-crypto-tests/src/Test/Crypto/Util.hs

```diff
@@ -242,6 +275,16 @@ prop_no_thunks !a = case unsafeNoThunks a of
     Nothing  -> property True
     Just msg -> counterexample (show msg) (property False)
 
+prop_no_thunks_IO_from :: NoThunks a => Lock -> (b -> IO a) -> b -> Property
+prop_no_thunks_IO_from lock mkX y = ioProperty . withLock lock $ do
+  x <- mkX y
+  return $ prop_no_thunks x
+
+prop_no_thunks_IO_with :: NoThunks a => Lock -> (Gen (IO a)) -> Property
+prop_no_thunks_IO_with lock mkX = ioProperty . withLock lock $ do
+  x <- join (generate mkX)
+  return $ prop_no_thunks x
```

Comment:
`generate` should never be used in actual tests. It is totally unreproducable. There is a better way to convert an `IO` action into a `property`
```suggestion
prop_no_thunks_IO_with :: NoThunks a => Lock -> Gen (IO a) -> Property
prop_no_thunks_IO_with lock mkX = forAll mkX (ioProperty . withLock lock . fmap prop_no_thunks)
```

Reply:
That should work, except there is no `instance Show (IO a)`, so we need to use `forAllBlind`, or else `forAllShow` with a naughty fake show function (`show . unsafePerformIO` or something like that).

However it seems to me that what we should really be doing here is just run `noThunks` in IO to begin with, instead of calling into `prop_no_thunks` which then uses `unsafeNoThunks`.

Reply:
And this led me to realize that the NoThunks tests on KES and DSIGNM keys and sigs were still commented out, so I fixed them, and re-enabled them, and, lo and behold, they actually found a bunch of thunks, so I had to go in and fix that, too.